main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala

```diff
@@ -27,16 +27,16 @@ object Unification {
   /**
     * Unify the two type variables `x` and `y`.
     */
-  private def unifyVars(x: Type.Var, y: Type.Var, renv: RigidityEnv)(implicit flix: Flix): Result[Substitution, UnificationError] = {
+  private def unifyVars(x: Type.Var, y: Type.Var, renv: RigidityEnv)(implicit flix: Flix): Result[(Substitution, List[(Type, Type)]), UnificationError] = {
```

Comment:
Perhaps we want something richer than (Type, Type). I am almost positive want that!

Reply:
I will let you do that in the next PR.